v3/src/components/slider/slider-component-handler.test.ts

```diff
@@ -0,0 +1,435 @@
+import { DIComponentInfo } from "../../data-interactive/data-interactive-types"
+import { diComponentHandler } from "../../data-interactive/handlers/component-handler"
+import { diGlobalHandler } from "../../data-interactive/handlers/global-handler"
+import { appState } from "../../models/app-state"
+import { toV3Id } from "../../utilities/codap-utils"
+import { isSliderModel } from "./slider-model"
+import { kSliderIdPrefix } from "./slider-registration"
+import { IDocumentContentModel } from "../../models/document/document-content"
+import { Logger } from "../../lib/logger"
+import { testGetComponent } from "../../data-interactive/handlers/component-handler-test-utils"
+import { getSnapshot } from "mobx-state-tree"
+import { kSliderTileType } from "./slider-defs"
+
+// The logger has to be initialized with a document before a new document
+// can be created.
+Logger.initializeLogger(appState.document)
+
+function createGlobalValue(name: string, value: number) {
+  const globalResult = diGlobalHandler.create!({}, {
+    type: "global",
+    name,
+    value,
+  })
+  expect(globalResult.success).toBe(true)
+  return name
+}
+
+describe("Slider ComponentHandler", () => {
+  const handler = diComponentHandler
+  let documentContent: IDocumentContentModel
+
+  beforeEach(async () => {
+    // Create a new document
+    await appState.setDocument({type: "CODAP"})
+    documentContent = appState.document.content!
+  })
+
+  test("create slider without a global", () => {
+    expect(documentContent.tileMap.size).toBe(0)
+
+    const result = handler.create!({}, { type: "slider" })
+    // TODO: is this correct? Without a global value, should the slider be created?
+    expect(result.success).toBe(true)
+    expect(documentContent.tileMap.size).toBe(1)
+
+    const resultValues = result.values as DIComponentInfo
+    const tile = documentContent.tileMap.get(toV3Id(kSliderIdPrefix, resultValues.id!))!
+    expect(tile).toBeDefined()
+    expect(isSliderModel(tile.content)).toBe(true)
+  })
+
+  test("cannot create slider with a globalValue that doesn't exist", () => {
+    expect(documentContent.tileMap.size).toBe(0)
+
+    const result = handler.create!({}, {
+      type: "slider",
+      globalValueName: "nonexistentGlobalValue",
+    })
+
+    expect(result.success).toBe(false)
+    if (result.success) {
+      throw new Error("Expected result to be unsuccessful")
+    }
```

Comment:
Are these extra checks necessary? Seems like if the result is anything other than `false`, the above check will fail and we'll bail on the test anyway.

Reply:
They narrow the type of the result so the code below that calls `result.values!.error` is allowed by typescript. I could get rid of the `expect(result.success).toBe(false)` so at least it isn't redundant. However in that case if the code changes and the test fails, what gets printed out by Jest isn't as easy to understand.

Reply:
This issue has more info on this: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/41179
But what isn't mentioned is the impact using the helper functions have on the Jest output when the tests fails.